### PR TITLE
Mock auth playwright

### LINKED ISSUE #60: [FEATURE] Playwight mock authentication
### Description

Test authenticated (logged in pages) with Playwright

### Screenshots

_No response_

### Additional information

This was done with BioDrop https://github.com/EddieHubCommunity/BioDrop/blob/main/tests/setup/auth.js

Note: in BioDrop another ORM was used

### PR DESCRIPTION
**Pull request by eddiejaoude**
## Fixes Issue

closes #60 

Original fork PR https://github.com/EddieHubCommunity/HealthCheck/pull/64

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -8,13 +8,30 @@ jobs:
   test:
     timeout-minutes: 60
     runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:16.3-alpine
+        env:
+          POSTGRES_DB: POSTGRES_DB
```
Comment (eddiejaoude) [inline .github/workflows/playwright.yml:15]:
```suggestion
          POSTGRES_DB: healthcheck
```

**Comment by amandamartin-dev** [comment]
Noticed this warnign from msw - haven't read up on it to know if we should definitely handle it but fyi in icymi

"[WebServer] [MSW] Warning: intercepted a request without a matching request handler:

  • GET https://github.com/EddieHubCommunity/HealthCheck/assets/[62](https://github.com/EddieHubCommunity/HealthCheck/actions/runs/10134518113/job/28020843158?pr=76#step:8:63)4760/c2fba520-d426-4c90-bce1-17fe9fb00041

If you still wish to intercept this unhandled request, please create a request handler for it.
Read more: https://mswjs.io/docs/getting-started/mocks"

Otherwise I didn't see anything that jumped out at me, looks pretty good.  Will take another look again in the am though

**Comment by dan-mba** [comment]
> Noticed this warnign from msw - haven't read up on it to know if we should definitely handle it but fyi in icymi
> 
> "[WebServer] [MSW] Warning: intercepted a request without a matching request handler:
> 
> • GET [https://github.com/EddieHubCommunity/HealthCheck/assets/[62](https://github.com/EddieHubCommunity/HealthCheck/actions/runs/10134518113/job/28020843158?pr=76#step:8:63)4760/c2fba520-d426-4c90-bce1-17fe9fb00041](https://github.com/EddieHubCommunity/HealthCheck/assets/%5B62%5D(https://github.com/EddieHubCommunity/HealthCheck/actions/runs/10134518113/job/28020843158?pr=76#step:8:63)4760/c2fba520-d426-4c90-bce1-17fe9fb00041)
> 
> If you still wish to intercept this unhandled request, please create a request handler for it. Read more: https://mswjs.io/docs/getting-started/mocks"
> 
> Otherwise I didn't see anything that jumped out at me, looks pretty good. Will take another look again in the am though

The unhandled requests seem to be for Google fonts & GitHub static assets.
This warning can be disabled, [see this discussion](https://github.com/mswjs/msw/discussions/1589#discussioncomment-5579717)

**Comment by eddiejaoude** [comment]
Thanks @amandamartin-dev and @dan-mba great points 👍

**Comment by eddiejaoude** [comment]
> This warning can be disabled

I think it might be better to leave the warnings because if we forget to mock one, we won't notice it

**Comment by dan-mba** [comment]
> > This warning can be disabled
> 
> I think it might be better to leave the warnings because if we forget to mock one, we won't notice it

I think using a selective bypass for specific domains would get completely irrelevant messages out of the log that may obscure real issues.
```js
server.listen({
  onUnhandledRequest(request, print) {
    const unmocked = ['googleapis.com','gstatic.com'];
    if (unmocked.some(host => request.url.hostname.includes(host))) {
      return;
    }

    // Print the regular MSW unhandled request warning otherwise.
    print.warning();
  }
});
```

**Comment by eddiejaoude** [comment]
Great point, I will give that a try, thank you 👍

**Comment by eddiejaoude** [comment]
Thanks everyone for the support, I have now made those changes 🎉 .

I think we should merge this and add more tests in another PR, as this PR is getting conflicts and very big

**Comment by dan-mba** [comment]
The following lines should be removed from `playwright.yml` as they aren't needed with the github reporter.
```yml
- uses: actions/upload-artifact@v4
  if: always()
  with:
    name: playwright-report
    path: playwright-report/
    retention-days: 30
```

They are generating a warning.

**Comment by eddiejaoude** [comment]
> The following lines should be removed from `playwright.yml` as they aren't needed with the github reporter.
> 
> ```yaml
> - uses: actions/upload-artifact@v4
>   if: always()
>   with:
>     name: playwright-report
>     path: playwright-report/
>     retention-days: 30
> ```
> 
> They are generating a warning.

WOW 😱  great spot @dan-mba 💪

**Pull request merged**